crates/wiggle/generate/src/funcs.rs

```diff
@@ -73,36 +73,55 @@ fn _define_func(
         },
     );
 
-    let asyncness = if settings.get_async(&module, &func).is_sync() {
-        quote!()
-    } else {
-        quote!(async)
-    };
     let mod_name = &module.name.as_str();
     let func_name = &func.name.as_str();
-    (
-        quote! {
+    if settings.get_async(&module, &func).is_sync() {
+        (
+            quote!(
+                #[allow(unreachable_code)] // deals with warnings in noreturn functions
+                pub fn #ident(
+                    ctx: &mut (impl #(#bounds)+*),
+                    memory: &dyn #rt::GuestMemory,
+                    #(#abi_params),*
+                ) -> Result<#abi_ret, #rt::Trap> {
+                    use std::convert::TryFrom as _;
+                    let _span = #rt::tracing::span!(
+                        #rt::tracing::Level::TRACE,
+                        "wiggle abi",
+                        module = #mod_name,
+                        function = #func_name
+                    );
+                    _span.in_scope(|| {
+                      #body
+                    })
+              }
+            ),
+            bounds,
+        )
+    } else {
+        (
+            quote!(
             #[allow(unreachable_code)] // deals with warnings in noreturn functions
```

Comment:
I think the indentation may be a bit off here and below? (to match the indentation above)

Reply:
Yeah cargo fmt can't really handle things inside macros. I think I got it fixed up in the most recent commit as well.